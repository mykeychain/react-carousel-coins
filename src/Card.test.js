import React from "react";
import { render } from "@testing-library/react";
import Card from "./Card";

const testProps = {
            caption: "test-caption",
            src: "test-src",
            currNum: "test-currNum",
            totalNum: "test-totalNum"
}

it("card renders without crashing", function() {
  render(<Card {...testProps}/>);
});

it("works when you click on the right arrow", function() {
  const { container } = render(<Card {...testProps}/>);

  // expect the first image to show, but not the second
  expect(
    container.querySelector('img[src="test-src"]')
  ).toBeInTheDocument();
});

it("matches snapshot", function() {
  const { container } = render(<Card {...testProps}/>);
  expect(container).toMatchSnapshot();
})