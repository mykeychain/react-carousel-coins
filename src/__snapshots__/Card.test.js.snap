// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      test-caption
    </h4>
    <img
      alt="test-caption"
      class="Card-image"
      src="test-src"
    />
    <small
      class="Card-small"
    >
      Image 
      test-currNum
       of 
      test-totalNum
      .
    </small>
  </div>
</div>
`;
